File: examples/colors.go
// =============================================================================
// Go Color Formats Example
// nvim-colorpicker detects and can replace all these formats
// =============================================================================

package colors

import (
	"image/color"
)

// Standard library color.RGBA (premultiplied alpha)
var (
	Primary    = color.RGBA{98, 0, 238, 255}
	Secondary  = color.RGBA{3, 218, 198, 255}
	Background = color.RGBA{18, 18, 18, 255}
	Surface    = color.RGBA{30, 30, 30, 255}
	Error      = color.RGBA{207, 102, 121, 255}
	Success    = color.RGBA{102, 187, 106, 255}
)

// Basic colors
var (
	Red     = color.RGBA{255, 0, 0, 255}
	Green   = color.RGBA{0, 255, 0, 255}
	Blue    = color.RGBA{0, 0, 255, 255}
	Yellow  = color.RGBA{255, 255, 0, 255}
	Cyan    = color.RGBA{0, 255, 255, 255}
	Magenta = color.RGBA{255, 0, 255, 255}
	White   = color.RGBA{255, 255, 255, 255}
	Black   = color.RGBA{0, 0, 0, 255}
)

// Semi-transparent colors
var (
	Overlay   = color.RGBA{0, 0, 0, 128}
	Highlight = color.RGBA{98, 0, 238, 64}
	Glass     = color.RGBA{255, 255, 255, 26}
	Shadow    = color.RGBA{0, 0, 0, 153}
)

// color.NRGBA (non-premultiplied alpha) - more intuitive for UI work
var (
	OverlayN   = color.NRGBA{0, 0, 0, 128}
	HighlightN = color.NRGBA{98, 0, 238, 64}
	GlassN     = color.NRGBA{255, 255, 255, 26}
	ShadowN    = color.NRGBA{0, 0, 0, 153}
)

// Material Design colors
var MaterialColors = struct {
	Red    color.RGBA
	Pink   color.RGBA
	Purple color.RGBA
	Blue   color.RGBA
	Cyan   color.RGBA
	Green  color.RGBA
	Yellow color.RGBA
	Orange color.RGBA
}{
	Red:    color.RGBA{244, 67, 54, 255},
	Pink:   color.RGBA{233, 30, 99, 255},
	Purple: color.RGBA{156, 39, 176, 255},
	Blue:   color.RGBA{33, 150, 243, 255},
	Cyan:   color.RGBA{0, 188, 212, 255},
	Green:  color.RGBA{76, 175, 80, 255},
	Yellow: color.RGBA{255, 235, 59, 255},
	Orange: color.RGBA{255, 152, 0, 255},
}

// Dark theme colors
var DarkTheme = struct {
	Background color.RGBA
	Surface    color.RGBA
	Primary    color.RGBA
	Accent     color.RGBA
	Text       color.RGBA
	TextMuted  color.NRGBA
}{
	Background: color.RGBA{26, 26, 46, 255},
	Surface:    color.RGBA{22, 33, 62, 255},
	Primary:    color.RGBA{15, 52, 96, 255},
	Accent:     color.RGBA{233, 69, 96, 255},
	Text:       color.RGBA{234, 234, 234, 255},
	TextMuted:  color.NRGBA{170, 175, 191, 170},
}

// Game palette
var GamePalette = struct {
	Player      color.RGBA
	Enemy       color.RGBA
	Collectible color.RGBA
	Obstacle    color.RGBA
	Health      color.RGBA
	Mana        color.RGBA
}{
	Player:      color.RGBA{79, 195, 247, 255},
	Enemy:       color.RGBA{239, 83, 80, 255},
	Collectible: color.RGBA{255, 213, 79, 255},
	Obstacle:    color.RGBA{120, 144, 156, 255},
	Health:      color.RGBA{102, 187, 106, 255},
	Mana:        color.RGBA{171, 71, 188, 255},
}

// Gradient colors
func SunsetGradient() []color.RGBA {
	return []color.RGBA{
		{255, 87, 34, 255},
		{255, 152, 0, 255},
		{255, 235, 59, 255},
	}
}

func OceanGradient() []color.RGBA {
	return []color.RGBA{
		{0, 188, 212, 255},
		{33, 150, 243, 255},
		{63, 81, 181, 255},
	}
}

// Helper to create color from hex
func FromHex(hex uint32) color.RGBA {
	return color.RGBA{
		R: uint8((hex >> 16) & 0xFF),
		G: uint8((hex >> 8) & 0xFF),
		B: uint8(hex & 0xFF),
		A: 255,
	}
}

// Hex color constants
const (
	HexRed    = 0xFF5722
	HexPink   = 0xE91E63
	HexPurple = 0x9C27B0
	HexBlue   = 0x2196F3
	HexCyan   = 0x00BCD4
	HexGreen  = 0x4CAF50
	HexYellow = 0xFFEB3B
	HexOrange = 0xFF9800
)

File: examples/colors.cs
// =============================================================================
// C#/Unity Color Formats Example
// nvim-colorpicker detects and can replace all these formats
// =============================================================================

using UnityEngine;
using System.Drawing; // For WPF/WinForms Color

namespace ColorExamples
{
    // Unity Color32 (bytes 0-255) - most common for game dev
    public static class GameColors
    {
        public static readonly Color32 Player = new Color32(79, 195, 247, 255);
        public static readonly Color32 Enemy = new Color32(239, 83, 80, 255);
        public static readonly Color32 Collectible = new Color32(255, 213, 79, 255);
        public static readonly Color32 Obstacle = new Color32(120, 144, 156, 255);
        public static readonly Color32 Health = new Color32(102, 187, 106, 255);
        public static readonly Color32 Mana = new Color32(171, 71, 188, 255);
        public static readonly Color32 UI_Text = new Color32(255, 255, 255, 255);
        public static readonly Color32 UI_Shadow = new Color32(0, 0, 0, 128);
    }

    // Unity Color (floats 0-1) with 'f' suffix
    public static class ThemeColors
    {
        public static readonly Color Primary = new Color(0.384f, 0.000f, 0.933f);
        public static readonly Color Secondary = new Color(0.012f, 0.855f, 0.776f);
        public static readonly Color Background = new Color(0.071f, 0.071f, 0.071f);
        public static readonly Color Surface = new Color(0.118f, 0.118f, 0.118f);
        public static readonly Color Error = new Color(0.812f, 0.400f, 0.475f);
        public static readonly Color Success = new Color(0.400f, 0.733f, 0.416f);
    }

    // Unity Color with alpha (floats)
    public static class TransparentColors
    {
        public static readonly Color Overlay = new Color(0.000f, 0.000f, 0.000f, 0.50f);
        public static readonly Color Highlight = new Color(0.384f, 0.000f, 0.933f, 0.25f);
        public static readonly Color GlassEffect = new Color(1.000f, 1.000f, 1.000f, 0.10f);
        public static readonly Color FadeOut = new Color(0.071f, 0.071f, 0.071f, 0.00f);
    }

    // WPF/WinForms Color.FromArgb (a, r, g, b)
    public static class WpfColors
    {
        public static readonly System.Drawing.Color Primary = System.Drawing.Color.FromArgb(255, 98, 0, 238);
        public static readonly System.Drawing.Color Secondary = System.Drawing.Color.FromArgb(255, 3, 218, 198);
        public static readonly System.Drawing.Color Background = System.Drawing.Color.FromArgb(255, 18, 18, 18);
        public static readonly System.Drawing.Color SemiTransparent = System.Drawing.Color.FromArgb(128, 0, 0, 0);
    }

    // WPF Color.FromRgb (r, g, b)
    public static class SimpleColors
    {
        public static readonly System.Drawing.Color Red = System.Drawing.Color.FromRgb(244, 67, 54);
        public static readonly System.Drawing.Color Green = System.Drawing.Color.FromRgb(76, 175, 80);
        public static readonly System.Drawing.Color Blue = System.Drawing.Color.FromRgb(33, 150, 243);
        public static readonly System.Drawing.Color Yellow = System.Drawing.Color.FromRgb(255, 235, 59);
    }

    // Dark theme for UI
    public static class DarkTheme
    {
        public static readonly Color32 Background = new Color32(26, 26, 46, 255);
        public static readonly Color32 Surface = new Color32(22, 33, 62, 255);
        public static readonly Color32 Primary = new Color32(15, 52, 96, 255);
        public static readonly Color32 Accent = new Color32(233, 69, 96, 255);
        public static readonly Color32 Text = new Color32(234, 234, 234, 255);
        public static readonly Color32 TextMuted = new Color32(170, 175, 191, 170);
    }

    // Particle system colors
    public class ParticleColors
    {
        public Color FireStart = new Color(1.000f, 0.400f, 0.000f, 1.00f);
        public Color FireEnd = new Color(1.000f, 0.000f, 0.000f, 0.00f);
        public Color SmokeStart = new Color(0.300f, 0.300f, 0.300f, 0.80f);
        public Color SmokeEnd = new Color(0.100f, 0.100f, 0.100f, 0.00f);
        public Color MagicStart = new Color(0.733f, 0.533f, 0.969f, 1.00f);
        public Color MagicEnd = new Color(0.400f, 0.200f, 0.800f, 0.00f);
    }
}
